core/lib/spree/bus.rb

```diff
@@ -0,0 +1,19 @@
+require 'omnes'
+
+module Spree
+  # Global [Omnes](https://github.com/nebulab/omnes) bus.
+  #
+  # This is used for internal events, while host applications are also able to
+  # use it.
+  #
+  # It has some modifications to support internal usage of the legacy event
+  # system {see Spree::AppConfiguration#use_legacy_events}.
+  Bus = Omnes::Bus.new
+  Bus.define_singleton_method(:publish) do |*args, **kwargs, &block|
+    if Spree::Config.use_legacy_events
+      Spree::Event.fire(*args, **kwargs, &block)
+    else
+      super(*args, **kwargs, caller_location: caller_locations(1)[0], &block)
```

Comment:
I was a bit puzzled as to why we are overriding the `caller_location` here. Would it be worth leaving a comment that this is intended to omit this from the stack trace, or just dropping that in order to ease debugging for anyone who is getting the new bus implementation and trying to track down why?

Reply:
I went with the addition of a comment. We do need to go one level higher in the stack. Otherwise, it'll be useless for debugging, as all events will look as published from this line (see https://github.com/nebulab/omnes/blob/bb73fce75414816e304a7402efbea655b82c857b/lib/omnes/bus.rb#L187)

Reply:
Thanks for clarifying this @waiting-for-dev, your decision here makes a lot more sense now.